metricflow-semantics/metricflow_semantics/model/semantics/semantic_model_lookup.py

```diff
@@ -241,17 +242,23 @@ def _add_semantic_model(self, semantic_model: SemanticModel) -> None:
             semantic_models_for_dimension = self._dimension_index.get(dim.reference, []) + [semantic_model]
             self._dimension_index[dim.reference] = semantic_models_for_dimension
 
+            assert StructuredLinkableSpecName.from_name(dim.name).is_element_name, (
+                f"Dimension name `{dim.name}` contains annotations, but this name should be the plain element name "
+                "from the original model. This should have been blocked by validation!"
+            )
+
+            # TODO: Construct these specs correctly. All of the time dimension specs have the default granularity
             self._dimension_ref_to_spec[dim.time_dimension_reference or dim.reference] = (
```

Comment:
This dict seems pretty useless since the specs don't have any new info compared to the references 🤔 I wonder if we can remove it.

Reply:
This is unclear to me. It's used in conversion metrics somehow, so maybe we can replace it over there?